crates/ruff_python_parser/src/parser/mod.rs

```diff
@@ -821,7 +836,14 @@ impl RecoveryContextKind {
             RecoveryContextKind::SetElements | RecoveryContextKind::DictElements => {
                 p.at(TokenKind::Rbrace)
             }
-            RecoveryContextKind::TupleElements => p.at(TokenKind::Rpar),
+            RecoveryContextKind::TupleElements(parenthesized) => {
+                if parenthesized.is_yes() {
+                    // TODO(dhruvmanila): Confirm if this is ok
+                    p.at(TokenKind::Rpar)
+                } else {
+                    p.at_sequence_end()
+                }
+            }
```

Comment:
I think we can just use the `at_sequence_end` even for parenthesized tuple which might help in error recovery. I'll need to test this out further but it shouldn't block this PR.